addons/web/controllers/main.py

```diff
@@ -1,1992 +1,55 @@
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-import base64
-import copy
-import datetime
-import functools
-import hashlib
-import io
-import itertools
-import json
-import logging
-import operator
-import os
-import re
-import sys
-import tempfile
-import unicodedata
 import warnings
-from collections import OrderedDict, defaultdict
-
-import babel.messages.pofile
-import werkzeug
-import werkzeug.exceptions
-import werkzeug.utils
-import werkzeug.wrappers
-import werkzeug.wsgi
-from lxml import etree, html
-from markupsafe import Markup
-from werkzeug.urls import url_encode, url_parse, iri_to_uri
-
-import odoo
-import odoo.modules.registry
 from odoo import http
-from odoo.api import call_kw
-from odoo.modules import get_resource_path, module, get_manifest
-from odoo.exceptions import AccessError, UserError, AccessDenied
-from odoo.http import content_disposition, request
-from odoo.models import check_method_name
-from odoo.service import db, dispatch_rpc, security
-from odoo.tools import html_escape, pycompat, ustr, apply_inheritance_specs, lazy_property, float_repr, osutil
-from odoo.tools.mimetypes import guess_mimetype
-from odoo.tools.misc import str2bool, xlsxwriter, file_open, file_path
-from odoo.tools.safe_eval import safe_eval, time
-from odoo.tools.translate import _
-
-from odoo.addons.base.models.ir_qweb import render as qweb_render
-
-
-_logger = logging.getLogger(__name__)
-
-CONTENT_MAXAGE = http.STATIC_CACHE_LONG  # menus, translations, static qweb
-
-DBNAME_PATTERN = '^[a-zA-Z0-9][a-zA-Z0-9_.-]+$'
-
-COMMENT_PATTERN = r'Modified by [\s\w\-.]+ from [\s\w\-.]+'
-
-def none_values_filtered(func):
-    @functools.wraps(func)
-    def wrap(iterable):
-        return func(v for v in iterable if v is not None)
-    return wrap
-
-def allow_empty_iterable(func):
-    """
-    Some functions do not accept empty iterables (e.g. max, min with no default value)
-    This returns the function `func` such that it returns None if the iterable
-    is empty instead of raising a ValueError.
-    """
-    @functools.wraps(func)
-    def wrap(iterable):
-        iterator = iter(iterable)
-        try:
-            value = next(iterator)
-            return func(itertools.chain([value], iterator))
-        except StopIteration:
-            return None
-    return wrap
-
-OPERATOR_MAPPING = {
-    'max': none_values_filtered(allow_empty_iterable(max)),
-    'min': none_values_filtered(allow_empty_iterable(min)),
-    'sum': sum,
-    'bool_and': all,
-    'bool_or': any,
+from odoo.tools import lazy
+from odoo.addons.web.controllers import (
+    action, binary, database, dataset, export, home, report, session,
+    utils, view, webclient,
+)
+
+_MOVED_TO_MAP = {
```

Comment:
Thanks for maintaining the compatibility! 👍👍

I'm wondering: is the `_MOVED_TO_MAP` + module-level `getattr` really better than a plain list of assignations? It has slightly less redundancy, but overall it feels less readable, and the `getattr` magic could backfire / shadow things, no?


```
_get_login_redirect_url = utils._get_login_redirect_url
_local_web_translations = utils._local_web_translations
...
```